resources/compute_virtual_machines.go

```diff
@@ -0,0 +1,608 @@
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ExtendedLocation.Name"),
+			},
+			{
+				Name:        "extended_location_type",
+				Description: "The type of the extended location Possible values include: 'EdgeZone'",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ExtendedLocation.Type"),
+			},
+			{
+				Name:        "id",
+				Description: "Resource Id",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ID"),
+			},
+			{
+				Name:        "name",
+				Description: "Resource name",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "type",
+				Description: "Resource type",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "location",
+				Description: "Resource location",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "tags",
+				Description: "Resource tags",
+				Type:        schema.TypeJSON,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:         "azure_compute_virtual_machine_win_config_rm_listeners",
+				Description:  "WinRMListener describes Protocol and thumbprint of Windows Remote Management listener",
+				Resolver:     fetchComputeVirtualMachineWinConfigRmListeners,
+				AlwaysDelete: true,
+				Columns: []schema.Column{
+					{
+						Name:        "virtual_machine_cq_id",
+						Description: "Unique ID of azure_compute_virtual_machines table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "virtual_machine_id",
+						Description: "ID of azure_compute_virtual_machines table (FK)",
+						Type:        schema.TypeString,
+						Resolver:    schema.ParentResourceFieldResolver("id"),
+					},
+					{
+						Name:        "protocol",
+						Description: "Specifies the protocol of WinRM listener <br><br> Possible values are: <br>**http** <br><br> **https** Possible values include: 'HTTP', 'HTTPS'",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "certificate_url",
+						Description: "This is the URL of a certificate that has been uploaded to Key Vault as a secret For adding a secret to the Key Vault, see [Add a key or secret to the key vault](https://docsmicrosoftcom/azure/key-vault/key-vault-get-started/#add) In this case, your certificate needs to be It is the Base64 encoding of the following JSON Object which is encoded in UTF-8: <br><br> {<br>  \"data\":\"<Base64-encoded-certificate>\",<br>  \"dataType\":\"pfx\",<br>  \"password\":\"<pfx-file-password>\"<br>}",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("CertificateURL"),
+					},
+				},
+			},
+			{
+				Name:         "azure_compute_virtual_machine_secrets",
+				Description:  "VaultSecretGroup describes a set of certificates which are all in the same Key Vault",
+				Resolver:     fetchComputeVirtualMachineSecrets,
+				AlwaysDelete: true,
```

Comment:
upgrade to RC5 and you can avoid this, also running always delete without a filter won't work
Also the PK here would be  source_vault_id